Experiments blogpost

neilkakkar opened the pull request:
## Changes

*Please describe.*

*Add screenshots or screen recordings for visual / UI-focused changes.*

## Checklist
- [ ] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [ ] Feature names are in [title case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case)
- [ ] Words are spelled using American English
- [ ] I have checked out our [style guide](https://github.com/PostHog/posthog.com/blob/master/STYLEGUIDE.md)

neilkakkar commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,106 @@
+
+(TK: well, all 4 params are dependent on time in a way, soooo.... this is a bit confusing)
+
+## Choosing the right metric is hard
```
the next 3 headings flow smooth AF into each other. Definitely want to keep these 3 together, but the 2 on top don't seem to fit as well. Hmm.

neilkakkar commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,106 @@
+
+We recently launched Experiments, and there's a ton we learned about running successful experiments. That's what this post is about.
+
+## The Peeking Problem is poorly named
```
definitely want to keep this, maybe as 4th, last point? Or as first as is.

neilkakkar commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,106 @@
+[^1]: If you're looking for how we calculate this, see the user guide (TK: link)
+
+
+## The kind of experiment you run determines how much data you need
```
perhaps delete section completely

neilkakkar commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,106 @@
+
+Multi armed bandits? -> No, skip.
+
+8. Clinical trials are very different from website changes -->>> hmm condense this. What am I trying to say?
```
Hmm, not sure I want to keep this, but can become decent companion to first point. Actually yes, I like that. Peeking, then nature of web products, then the 3 in flow.

neilkakkar commented:
@andyvan-ph still an early draft, but decent enough for a first read through. Does this sound compelling/exciting enough, or is it too basic / too meh ?

Still trying to figure out which sections to keep and delete, and if I want more sections.

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,106 @@
+title: What launching Experimentation taught us about running effective A/B tests
+---
+
+PostHog started off with simple product analytics. You want some data analysed, so you ingest this data. PostHog then lets you visualise this, slice it over multiple axes, and helps you answer any product questions you have.
```
Agreed. As this is part of launch week, we should start with the thing we're launching.

neilkakkar replied:
Agreed!

andyvan-ph reviewed (commented):
A few thoughts. 

- I agree with @liyiy, the intro should start with introducing Experimentation in a top-level way.

- I think the current opening two sections (after the intro) could be one shorter section that provide an overview of what our feature set is and our philosophy around the peeking problem, and why we've built it the way we did. Some before and after screenshots of what Experiments look like while they're running vs after they're complete would be nice here.

- After that, just keep building on what you've done which looks good to me. We just need to tighten up the top half so we can get the 'meat' a bit quicker.

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,106 @@
+---
+title: What launching Experimentation taught us about running effective A/B tests
```
On title, I'm imagining something like this:

Introducing Experimentation (and what it taught us about running effective A/B tests)
Introducing Experimentation and x things we learnt about running effective A/B tests

neilkakkar replied:
Makes sense.

Introducing Experimentation and 3 things we learnt about running effective A/B tests

andyvan-ph replied:
Going to stick with your title

neilkakkar commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,106 @@
+
+## The Peeking Problem is poorly named
+
+The strawman pop-sci version of the peeking problem goes something like: "You shouldn't look at experiment results while the experiment is running because that can lead to you ending experiments early, when the data is skewed in favour of one variant, thanks to random chance."
```
Nope!

neilkakkar replied:
(I mean, good question, I decided to get rid of that section, its probs not for this post)

neilkakkar commented:
Good suggestions!

I deleted the peeking problem + amount of data required section, incorporated your feedback for the opening, and added a "bonus" thing we learned at the end, which I _think_ rounds out the post nicely.

Thanks for the prelim feedback, @liyiy and @andyvan-ph , that helped iterate quicker :)

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+hideAnchor: true
+author: ["neil-kakkar"]
+featuredImage: ../images/blog/posthog-engineering-blog.png
+featuredImageType: full
```
```suggestion
featuredImageType: full
categories: ["Product updates", "Launch week"]
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+We just launched our Experimentation suite, and there's a ton we learned about running successful experiments.
+
+It was a no brainer product decision: Since you're already analysing your data in PostHog, and you're already using feature flags to roll out new features, why not have the capability to test how well these features are doing?
```
```suggestion
It was a no brainer product decision: Since you're already analysing your data in PostHog, and you're already using [Feature Flags](/product/feature-flags) to roll out new features, why not have the capability to test how well these features are doing?
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+![PostHog - Experiment Creation](../images/blog/experimentation/experiment-creation.png)
+
+Thanks to feature flags, you can then easily validate whether each variant looks good, launch your experiment, and wait for data to come in. Once data starts trickling in, we run a Bayesian analysis to give a probability for each variant being the best, a graph of how things are looking for each variant, and whether the results are significant or not.
```
```suggestion
Thanks to Feature Flags, you can then easily validate whether each variant looks good, launch your experiment, and wait for data to come in. Once data starts trickling in, we run a Bayesian analysis to give a probability for each variant being the best, a graph of how things are looking for each variant, and whether the results are significant or not.
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+![PostHog - Experiment Results](../images/blog/experimentation/experiment-results.png)
+
+Well, that's enough about how experiments work. If you're interested in the technical details, check out the user guide. (TK: link) For now, let's get into three interesting things we learned about running successful experiments.
```
```suggestion
Well, that's enough about how experiments work. If you're interested in the technical details, check out the [Experimentation user guide](/docs/user-guides/experimentation). For now, let's get into three interesting things we learned about running successful experiments.
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+Well, that's enough about how experiments work. If you're interested in the technical details, check out the user guide. (TK: link) For now, let's get into three interesting things we learned about running successful experiments.
+
+
+## Choosing the right metric is hard
```
```suggestion
## 1. Choose your metric carefully
```

andyvan-ph replied:
Saying it's hard feels a bit reductive

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+Which one do you choose?
+
+Note that you're choosing the total interactions here, not unique interactions. Which means if one person clicks on the funnel three times, that counts as 3 interactions for either metric, as it should.
```
```suggestion
Note that you're choosing the total interactions here, not unique interactions. Which means if one person clicks on the funnel three times, that counts as three interactions for either metric, as it should.
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+[^2]: Depends a lot on how big and mature your product is, and how many sub features it has. If there's only one thing the app does, the local metric is the global metric.
+
+## Experiment results aren't set in stone
```
```suggestion
## 2. Experiment results aren't set in stone
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+## Experiment results aren't set in stone
+
+So, you just finished running the experiment above, and the results are in. Horizontal funnel layout had 1000 interactions, while vertical funnel layout had 1200. The results ended up being significant, with vertical funnel layout being 20% better.
```
```suggestion
So, you just finished running the experiment above, and the results are in. Horizontal funnel layout had 1,000 interactions, while vertical funnel layout had 1,200. The results ended up being significant, with vertical funnel layout being 20% better.
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+Do you blindly trust the data, or your intuition?
+
+Both have issues. Your intuition might be how you see the world, but not necessarily how people who use the product see the world. At the same time, what if there was a bug in the vertical layout implementation, which counted each interaction twice? Thus, the real number could've been 600, instead of 1200, which massively changes your product direction.
```
```suggestion
Both have issues. Your intuition might be how you see the world, but not necessarily how people who use the product see the world. At the same time, what if there was a bug in the vertical layout implementation, which counted each interaction twice? Thus, the real number could've been 600, instead of 1,200, which massively changes your product direction.
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+Both have issues. Your intuition might be how you see the world, but not necessarily how people who use the product see the world. At the same time, what if there was a bug in the vertical layout implementation, which counted each interaction twice? Thus, the real number could've been 600, instead of 1200, which massively changes your product direction.
+
+I'd recommend neither blindly trusting the data, nor your intuition. Experiments show you what is happening, but can't answer why. The real institutional knowledge comes with answering the why, and building up  an accurate model of who your users are, what they need, and how they interact with your product.
```
```suggestion
I'd recommend neither blindly trusting the data, nor your intuition. Experiments show you what is happening, but can't answer why. The real institutional knowledge comes with answering the why, and building up an accurate model of who your users are, what they need, and how they interact with your product.
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+I'd recommend neither blindly trusting the data, nor your intuition. Experiments show you what is happening, but can't answer why. The real institutional knowledge comes with answering the why, and building up  an accurate model of who your users are, what they need, and how they interact with your product.
+
+To answer the why, you need to talk through the causes. <a target='_blank' rel="noopener" href='https://neilkakkar.com/Bayes-Theorem-Framework-for-Critical-Thinking.html'> Create hypotheses about why this is happening</a> , watch user session recordings, and then make a call about what you want to do.
```
I see you Neil, plugging your own blog! 😆

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+That is, bring data to conversations, but also talk through causes.
+
+## Changing environments can invalidate experiment results
```
```suggestion
## 3. Changing environments can invalidate experiment results
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+## Changing environments can invalidate experiment results
+
+So, you've finally gotten results for the experiment above, and figured out why they're like this. Turns out, the vertical layout promotes interaction as it allows users to see all steps of the funnel in one go, click on the steps that seem surprising, see the persons involved in that step, watch their recordings, etc. etc. While the horizontal is a bit more frustrating to see all this informatio ...
```
```suggestion
So, you've finally gotten results for the experiment above, and figured out why they're like this. Turns out, the vertical layout promotes interaction – it allows users to see all steps of the funnel in one go, click on the steps that seem surprising, see the persons involved in that step, watch their recordings, etc. The horizontal layout, meanwhile, is a bit more frustrating to see all this information at a glance, causing faster bounces.
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+That's a model that keeps on giving, even when things change.
+
+Let's say its now three months in the future, and you've done a design revamp. Horizontal bars are thinner now, while vertical bars are thicker. As a result, horizontal funnels fit in cleanly on screen, while vertical funnels don't.[^4]
```
```suggestion
Let's say it's now three months in the future, and you've done a design revamp. Horizontal bars are thinner now, while vertical bars are thicker. As a result, horizontal funnels fit in cleanly on screen, while vertical funnels don't.[^4]
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+[^4]: A bit contrived, but easier to keep this blogpost manageable without explaining the ins and outs of PostHog graphs :). Also, don't tell our designer, they'll scream if they hear of this design revamp.
+
+All this is trying to say is experiment results don't stand the test of time, and reiterates the importance of extracting a useful model out of your experiment results.
+
```
```suggestion
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,94 @@
+
+Thus, moving quickly trumps rigor in web product experiments.
+
+We built experiments with this in mind. It's a web product, built for products that move quickly. That's all for this post, we'd love to have you try Experiments out, and tell us what you learn :)
```
```suggestion
We built Experimentation with this in mind. It's a web product, built for products that move quickly. That's all for this post, we'd love to have you try Experiments out, and tell us what you learn. To do so, feel free to open an issue in [our Github repo](https://github.com/PostHog/posthog), give us a shout in our [PostHog community Slack channel](/slack), or join us directly for a [call](https://calendly.com/posthog-feedback) with our Product & Engineering team.
```

andyvan-ph commented:
Ok, this is shaping up nicely.

I feel like it's shame we've lost your bit about the peeking problem entirely. Perhaps you could work this back into the last bonus section? I feel like it fits well with the jist of your point there - i.e. it's ok to peek if you don't draw conclusions too early, and web products are flexible etc. I feel like it's nice scene setting for the way we think when we're building features.

neilkakkar commented:
Makes a lot of sense actually, it fits right in!

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,108 @@
+---
+date: 2022-03-21
```
```suggestion
date: 2022-03-23
```

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,108 @@
+
+![PostHog - Experiment Results](../images/blog/experimentation/experiment-results.png)
+
+Well, that's enough about how experiments work. If you're interested in the technical details, check out the [Experimentation user guide](/docs/user-guides/experimentation). For now, let's get into three interesting things we learned about running successful experiments.
```
```suggestion
It's a powerful tool for building great products, but that's enough about how Experimentation work for now. If you're interested in the technical details, check out the [Experimentation user guide](/docs/user-guides/experimentation).  

Instead, I'm going to share some of the secrets we've learned about running effective A/B tests so you can get the most out of this new feature.
```

neilkakkar commented:
@andyvan-ph I assume this is good to go, no other changes needed? :)

andyvan-ph commented on contents/blog/experiments.md:
```diff
@@ -0,0 +1,110 @@
+author: ["neil-kakkar"]
+featuredImage: ../images/blog/experiments.jpeg
+featuredImageType: full
+categories: ["Product updates", "Launch week"]
```
```suggestion
categories: ["Product updates", "Launch week", "Guides"]
```

charlescook-ph reviewed (approved):
Made a smattering of small tweaks for flow, US spelling and removing extraneous 'thus'.
